pkg/cmd/releasenotes/render.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (C) 2021 ScyllaDB
+
+package releasenotes
+
+import (
+	"fmt"
+	"html/template"
+	"io"
+	"regexp"
+	"strings"
+)
+
+const releaseNotesTemplate = `
+# Release notes for {{ .Release }}
+## Container images
+
+` + "```\ndocker.io/scylladb/scylla-operator:{{ .Release }}\n```" + `
+
+## Changes By Kind
+
+{{- range $section := .Sections }}
+  {{- if $section.PullRequests }}
+
+### {{ $section.Title }}
+
+    {{- range $pr := $section.PullRequests }}
+{{ FormatPR $pr }}
+    {{- end }}
+  {{- end }}
+{{- end }}
+
+---
+`
+
+var branchPrefixRe = regexp.MustCompile(`^(\[[^]].*])?(.*)`)
+
+func cleanupPRTitle(title string) (string, error) {
+	m := branchPrefixRe.FindStringSubmatch(strings.TrimSpace(title))
+	if len(m) != 3 {
+		return "", fmt.Errorf("%q PR title cannot be parsed", title)
+	}
+	return strings.TrimSpace(m[2]), nil
+}
+
+func prURL(pr PullRequest) string {
+	return fmt.Sprintf("https://github.com/%s/%s/pull/%d", pr.Repository.Owner.Login, pr.Repository.Name, pr.Number)
+}
+
+func userURL(login string) string {
+	return fmt.Sprintf("https://github.com/%s", login)
+}
+
+func formatPR(pr PullRequest) (string, error) {
+	title, err := cleanupPRTitle(string(pr.Title))
+	if err != nil {
+		return "", err
+	}
+	userLogin := pr.Author.Login
+	details := fmt.Sprintf("[#%d](%s),[@%s](%s)", pr.Number, prURL(pr), userLogin, userURL(string(userLogin)))
+	return fmt.Sprintf("* %s (%s)", title, details), nil
+}
+
+type section struct {
+	Title        string
+	PullRequests []PullRequest
+}
+
+type releaseNoteData struct {
+	Sections []section
+	Release  string
+}
+
+func renderReleaseNotes(out io.Writer, release string, pullRequests []PullRequest) error {
+	sections := []section{
+		{"API Changes", filterByLabelKinds(pullRequests, []string{"api-change"})},
```

Comment:
If we add a new label for kind/ we might miss it here. How about creating a map from kind->[]PullRequest for all of them, then a helper map to give the categories readable names and those that don't have a readable names go into "Others"? The other option would be to fail and explicitly require the new category to be mapped into Others or given a readable name by updating the release note tool but I think "Others" would be fine.

Reply:
uncategorized would go into others with a warning log